src/parser/lexer.rs

```diff
@@ -0,0 +1,140 @@
+use std::{iter::Peekable, str::Chars};
+
+use super::token::Token;
+use super::range::{Range, Position};
+
+/// The lexer state
+#[derive(Debug)]
+pub struct Lexer<'a> {
+    pub input: &'a str,
+    pub position: usize,
+    pub peekable: &'a mut Peekable<Chars<'a>>
```

Comment:
Why use a mutable reference here and not, e.g., store `Peekable` by value?

Reply:
Uuh Idk, it's something that I didnt think about too much in the kind-parser crate.

Reply:
I just follow the `kind2` parser lmao